src/Analysis/Ast/Impl/Extensions/AstExtensions.cs

```diff
@@ -60,9 +61,23 @@ public static Expression FindExpression(this PythonAst ast, SourceLocation locat
             return ast.CommentLocations[match].Column < location.Column;
         }
 
+        internal static bool HasComment(this PythonAst ast, PythonModule module, string comment, int lineNum) {
+            var match = Array.BinarySearch(ast.CommentLocations, new SourceLocation(1, lineNum));
+
+            // line is not a comment
+            if (match < 0 || match >= ast.CommentLocations.Length) {
+                return false;
+            }
+
+            var commentLoc = ast.CommentLocations[match];
+            string line = module.GetLine(commentLoc.Line);
+
+            return line.Contains(comment);
```

Comment:
I'm thinking this is too loose; if you're just doing `line.Contains("noqa")`, that's going to remove too much. Before I had said that you should really only be checking if a comment is exactly `"noqa"` (case insentitive), ignoring whitespace.

Reply:
You're right, I forgot about that again!